src/legacy/core_plugins/vis_type_markdown/public/markdown_fn.ts

```diff
@@ -17,34 +17,58 @@
  * under the License.
  */
 
-import { functionsRegistry } from 'plugins/interpreter/registries';
 import { i18n } from '@kbn/i18n';
+import { ExpressionFunction, KibanaDatatable, Render } from '../../interpreter/types';
 
-export const kibanaMarkdown = () => ({
-  name: 'markdownVis',
+const name = 'markdownvis';
+
+interface Arguments {
+  markdown: string;
+  fontSize: number;
+  openLinksInNewTab: boolean;
+}
+
+type Return = Promise<Render<any>>;
```

Comment:
We already know what value `Render` returns, so we should probably provide that to the generic here:
```suggestion
type Return = Promise<Render<{ visType: 'markdown'; visConfig: Arguments }>>;
```